inspire_classifier/api.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Classifier API."""
+
+from __future__ import absolute_import, division, print_function
+
+from flask import current_app
+from inspire_classifier.core.ml.models import (
+    Classifier,
+    LanguageModel
+)
+from inspire_classifier.core.preprocessor.preprocessor import (
+    generate_and_save_classifier_tokens,
+    generate_and_save_language_model_tokens,
+    map_and_save_tokens_to_ids_for_classifier,
+    map_and_save_tokens_to_ids_for_language_model,
+    split_and_save_data_for_language_model_and_classifier
+)
+import os
+
+
+def preprocess_and_save_data():
```

Comment:
Sanity checks is a good idea. I will add those. 
For the API, so I can make a function called 'train' on top of these functions and make a separate function out of it. As for the predict function, I can rename it. However, I think it would still be a good idea to keep the sub-functions which would make the train function.

Reply:
Maybe I should have been more explicit, but I was thinking of human-friendly errors rather than a bunch of assertions.